app/models/vm.rb

```diff
@@ -112,4 +114,60 @@ def remote_console_url=(url, user_id)
   def generic_custom_buttons
     CustomButton.buttons_for("Vm")
   end
+
+  def supported_consoles
+    {
+      :spice    => spice_support,
+      :vnc      => vnc_support,
+      :vmrc     => vmrc_support,
+      :webmks   => webmks_support,
+      :cockpit  => cockpit_support
+    }
+  end
+
+  private
+
+  def vnc_support
+    {
+      :visible => supports_vnc_console?,
+      :enabled => supports_launch_vnc_console?,
+      :message => unsupported_reason(:launch_vnc_console)
+    }
+  end
+
+  def webmks_support
+    {
+      :visible => supports_mks_console?,
+      :enabled => supports_launch_mks_console?,
+      :message => unsupported_reason(:launch_mks_console)
+    }
+  end
+
+  def vmrc_support
+    {
+      :visible => supports_vnc_console?,
+      :enabled => supports_launch_vmrc_console?,
+      :message => unsupported_reason(:launch_vmrc_console)
+    }
+  end
+
+  def spice_support
+    {
+      :visible => supports_spice_console?,
+      :enabled => supports_launch_spice_console?,
+      :message => unsupported_reason(:launch_spice_console)
+    }
+  end
+
+  def cockpit_support
```

Comment:
@bmclaughlin  since `supports_launch_cockpit?` was already on this branch, what do you think we should be returning here? for now, both visible and enabled will return `supports_launch_cockpit?`, but wanted to get your input.